spot_wrapper/calibration/calibration_util.py

```diff
@@ -0,0 +1,916 @@
+            from the pattern, so that camera frames facing the board may be sampled
+            after applying a distance and offset rotation.
+            Defaults to np.array([[1, 0, 0], [0, -1, 0], [0, 0, -1]]).
+        degree_offset_rotations (bool, optional): whether to use degree for the
+            sampling rotation parameters. Defaults to True.
+
+    Returns:
+        List[np.ndarray]: a list of 4x4 homogenous transforms to visit in the "principal" cameras
+            frame
+    """
+    if degree_offset_rotations:
+        x_axis_rots = [radians(deg) for deg in x_axis_rots]
+        y_axis_rots = [radians(deg) for deg in y_axis_rots]
+        z_axis_rots = [radians(deg) for deg in z_axis_rots]
+    translations = [tvec + R_board[:, 2] * dist for dist in distances]
+    R_base = R_board @ R_align_board_frame_with_camera
+
+    def euler_to_rotation_matrix(roll: float, pitch: float, yaw: float) -> np.ndarray:
+        """
+        Convert Euler angles to a rotation matrix using OpenCV, for the sake
+        of CLI convenience, conciseness, while not depending on an external transform lib.
+        """
+        R_x = cv2.Rodrigues(np.array([roll, 0, 0]))[0]
+        R_y = cv2.Rodrigues(np.array([0, pitch, 0]))[0]
+        R_z = cv2.Rodrigues(np.array([0, 0, yaw]))[0]
+        return R_z @ R_y @ R_x
+
+    poses = []
+    for t in translations:
+        for x_axis_rot in x_axis_rots:
+            for y_axis_rot in y_axis_rots:
+                for z_axis_rot in z_axis_rots:
+                    # Get the rotation matrix from Euler angles
+                    R_rot = euler_to_rotation_matrix(x_axis_rot, y_axis_rot, z_axis_rot)
+
+                    # Combine with the base rotation
+                    R_base_modified = R_base @ R_rot
+
+                    transform = np.eye(4)
+                    transform[:-1, -1] = t.reshape((3,))
+                    transform[:-1, :-1] = R_base_modified
+                    poses.append(transform)
+
+    logger.info(f"Calculated {len(poses)} relative viewpoints to visit relative to the target.")
+    return poses
+
+
+def create_calibration_save_folders(path: str, num_folders: int) -> None:
+    """
+    Create a folder hierarchy to record a calibration
+
+    Args:
+        path (str): The parent path
+        num_folders (int): How many cameras, a.k.a how many folder to create
+
+    Raises:
+        ValueError: Not possible to create the folders, or no path specified
+    """
+    if path is None:
+        raise ValueError("No path to save to :(")
+    else:
+        for idx in range(num_folders):
+            cam_path = os.path.join(path, str(idx))
+
+            logger.info(f"Creating image folder at {cam_path}")
+            os.makedirs(cam_path, exist_ok=True)
+        logger.info(f"Done creating {num_folders} folders.")
+
+
+def load_images_from_path(path: str) -> np.ndarray:
+    """
+    Load image dataset from path in a way that's compatible with multistereo_calibration_charuco.
```

Comment:
Can you maybe add a blurb about the expected format of the data set and the path just to make it clearer? I see path is more the folder path and each camera is stored in it's own directory? Something short and sweet like that could make it way quicker to understand whats going on when its out of context.

Reply:
In the readme, I have:

# Using the CLI Tool To Calibrate On an Existing Dataset
To use the CLI Tool, please ensure that you have one parent folder, 
where each camera has a folder under the parent (Numbered from 0 to N). Synchronized photos 
for each camera should appear in their respective directories, where matching photos have ids, ascending
upwards, starting from 0. The file structure should appear something like the following:
```
existing_dataset/
├── 0/
│   ├── 0.png # taken at viewpoint 0 
│   ├── 1.png
│   └── 2.png
├── 1/
│   ├── 0.png # taken at viewpoint 0
│   ├── 1.png
│   └── 2.png
├── 2/
│   ├── 0.png # taken at viewpoint 0
│   ├── 1.png
│   └── 2.png
```

To see all possible arguments for calibration, please run  ```python3 calibrate_multistereo_cameras_with_charuco_cli.py -h```.
    Many parameters such as board proportions and Agmruco dictionary are customizable.

If you'd like to register camera 1 to camera 0, and camera 2 to camera 0, you could do the following:
```
python3 calibrate_multistereo_cameras_with_charuco_cli.py --data_path ~/existing_dataset/ \
--result_path ~/existing_dataset/eye_in_hand_calib.yaml --photo_utilization_ratio 1 --stereo_pairs "[(1,0), (2, 0)]" \
--tag default --unsafe_tag_save
``` 


I can also add something here if you don't think the readme is sufficient

Reply:
Thats my bad for skimming the readme! It might be good to call our the readme though here: something like: dataset is expected to be in the format detailed in  the readme. Itll save bad people like me from missing it lol.